pyoxidizer/src/distutils/command/build_ext.py

```diff
@@ -483,7 +488,30 @@ def _filter_build_errors(self, ext):
             self.warn('building extension "%s" failed: %s' %
                       (ext.name, e))
 
+    def _pyoxidizer_restore_from_cache(self, ext):
+        d = get_extension_details(ext.name)
+        if d is None:
+            return False
+        is_cached = all((d["dist_name"] == self.distribution.get_name(),
+                         d["dist_version"] == self.distribution.get_version(),
+                         d["name"] == ext.name))
```

Comment:
need to check here all the other stuff too. ie. must check the dependent libraries, the list of compilation units, etc. no? all the fields from the extension details json

Reply:
Most of the stuff in the JSON are only known post compilation 🥺

Reply:
mmm i see that libraries is irrelevant actually as that's only used for linking SO. but list of compilation units and the other stuff i mentioned is definitely available pre-compilation (because its needed for compilation)

Reply:
also, list of libraries is also available pre-compilation (see `get_libraries()`).

Reply:
> get_libraries

The list of compilation units should not change between package versions

Reply:
some packages can have a setup.py that conditionally includes compilation units depending on environment

Reply:
I've added the source file list to the cache, but wondering if thats enough.
Maybe calculate a rolling hash on these files as well to ensure the contents of the source file is not changed? or is that an overkill.

Reply:
![image](https://user-images.githubusercontent.com/2903914/180651859-225280a5-f2ea-4c1c-aa03-f5fe5d2d7d16.png)

Reply:
if the contents of the files change, then the package version ought to change as well

Reply:
unless the package generates source files during `setup.py`. now _that_ would require checking. actually now that i think about it, it's likely that any package that uses cffi does that!

Reply:
@d3dave Hashing file contents now